master/internal/trial.go

```diff
@@ -381,6 +381,23 @@ func (t *trial) maybeAllocateTask() error {
 	}
 
 	restoredAllocation, err := t.maybeRestoreAllocation()
+
+	if restoredAllocation == nil {
+		launchWarnings, err := t.rm.ValidateResourcePoolAvailability(
+			&sproto.ValidateResourcePoolAvailabilityRequest{
+				Name:   t.config.Resources().ResourcePool(),
+				Slots:  t.config.Resources().SlotsPerTrial(),
+				TaskID: &t.taskID,
+			},
+		)
+		if len(launchWarnings) > 0 {
+			logrus.Warnf("task ID %v slots requested exceeds cluster capacity", t.taskID)
+		}
+		if err != nil {
```

Comment:
check this error first, just since we shouldn't be looking at function return values unless we check the error first